### DISCUSSION TITLE
Control multiple plugs with smartphone using an ESP32 ventral node (AP mode, no router)

### DISCUSSION THREAD

**Discussion by risoul** [arendst/Tasmota#12558] [Q&A] [answered]
For a project, I need to control multiple plugs (up to 5) from smartphone, WITHOUT router, only using AP.

What I imagined is to create a central ESP32 node. This node runs a ESPAsyncwebserver and must be able to do automated task on the plugs (close at x hour, open at x°F). We connect to this central node with smartphone via Wifi AP. Then we have an interface where we choose and connect the detected plugs + buttons to control them.

Is it possible to do that?

![Capture d’écran_2021-07-04_19-00-01](https://user-images.githubusercontent.com/10132802/124393303-19570480-dcfa-11eb-8936-db98d0ace54f.png)

**Comment by barbudor** [comment, marked as answer]
Everything is possible. 
Is it the best to do ? that's another question

Tasmota needs an AP to connect to as well as a NTP server to provide time (Tasmota doesn't not without time)
Instead of an ESP32 as the central node, I would advise an RasPiZero which can support
- AP
- NTP server (still will need a time reference like a RTC (Dallas chip for example) or a GPS
- MQTT Broker
- Automation system, dashboard, etc ...

Sincerely, not worth trying to get an ESP32 as central node when an RasPi can do a better job and be more flexible
And if the RPi AP is not enough just get a cheap one. Less than 20€ on Ali

**Reply by risoul** [reply]
Thank you for your fast answer, I would like to avoid Raspberry because of card corruption risk. The power supply is not garanted, and it is easy to crash raspberry in this conditions. It is why I prefer microcontroler based system like ESP because it must be "bulletproof".

From what I read, ESP32 support :
- AP (and even AP/STA at the same time)
- NTP : I will add an RTC to the ESP32, with manual setting for the time. It seems it can act as a NTP server.
- I do not need MQTT
- Dashboard not so difficult to do with ESPAsyncserver

What are for you the blocking points?

**Reply by barbudor** [reply]
Hi risoul
No blocking points. 
As I said totally achievable. There multiple ways of doing it, I would choose another one. But that purely personal preference.

Yes the ESP32 (even ESP8266) can act as a AP, as a NTP server, as a Web server but you will have to cook all by yourself into a single application.
Everytime you will want to do more on the dashboard, etc... you will have to modify the code and upload it again.
You may feel that you do not need MQTT because you may never have used it.
The best way of using Tasmota is over MQTT, not Web requests.

Still I would advise a true AP, even a cheap one, instead of using the ESP32.
AP on an ESP is ok for configuration portal but I don't know if it can handle many devices.

**Reply by risoul** [reply]
Thanks, happy to know it could work.

The idea came from there : https://lemariva.com/blog/2018/10/micropython-controlling-smart-power-outlets-using-an-esp32.

What do you mean by "true AP", a router? do you have an example link?

**Reply by barbudor** [reply]
A router connects to a main network (like your Internet Provider) and routes the packets between the maiun network and your local/private network. A router doesn't necessarily have Wifi and could be only Ethernet.
An pure access point offers a Wifi SSID and bridge the Wifi network to your local network.
Sometimes/often a router also includes an AP.

Simple one like the those https://www.aliexpress.com/store/group/Mini-Router/5100113_518508520.html
I still have a few old T-Link WR703Nthat I purchased for less than 20€ a few years ago.
ESP devices only needs good-old 2.4GHz. You don't need fancy features. 
So the the simplest, the better.

**Reply by risoul** [reply]
Ok will try, I have one repeater that can be configured as AP. Thanks
I also did one one ESP32, so I will try with or without external materials.
Thanks for the great work you did, Tasmota is just an amazing tool ;-)
